### PR TITLE
doc: Update dependencies.md

### PR DESCRIPTION
**Pull request by hebasto**
Update docs according to the recent changes in the code:
- #21209 (zlib)
- #21376 (libnatpmp)

### CODE REVIEW

File: doc/dependencies.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ These are the dependencies currently used by Bitcoin Core. You can find instruct
 | GCC |  | [7+](https://gcc.gnu.org/) (C++17 support) |  |  |  |
 | HarfBuzz-NG |  |  |  |  | [Yes](https://github.com/bitcoin/bitcoin/blob/master/depends/packages/qt.mk) |
 | libevent | [2.1.11-stable](https://github.com/libevent/libevent/releases) | [2.0.21](https://github.com/bitcoin/bitcoin/pull/18676) | No |  |  |
-| libnatpmp | [20150609](https://miniupnp.tuxfamily.org/files) |  | No |  |  |
+| libnatpmp | [4536032...](https://github.com/miniupnp/libnatpmp/tree/4536032ae32268a45c073a4d5e91bbab4534773a) |  | No |  |  |
```
[the remark this answers is not in the document]
Comment (hebasto) [reply doc/dependencies.md]: Thanks. Updated.

**Comment by hebasto** [comment]
Updated f168e0622d15694f565bcdd3ded3e441d8402aee -> bb3f79fbdcd792aff04b2bf111de13219cd9ad56 ([pr21435.01](https://github.com/hebasto/bitcoin/commits/pr21435.01) -> [pr21435.02](https://github.com/hebasto/bitcoin/commits/pr21435.02), [diff](https://github.com/hebasto/bitcoin/compare/pr21435.01..pr21435.02)):

- addressed @laanwj's [comment](https://github.com/bitcoin/bitcoin/pull/21435#discussion_r595400732)

**Comment by laanwj** [comment]
Thanks, LGTM now

File: doc/dependencies.md [hunk 2]
```diff
@@ -43,3 +43,4 @@ Some dependencies are not needed in all configurations. The following are some f
 
 #### Other
 * librsvg is only needed if you need to run `make deploy` on (cross-compilation to) macOS.
+* Not-Qt-bundled zlib is required to build the [DMG tool](../contrib/macdeploy/README.md#deterministic-macos-dmg-notes) from the libdmg-hfsplus project.
```
Comment (fanquake) [inline doc/dependencies.md:46]: This can just be zlib

**Review by fanquake — APPROVED** [review]
ACK bb3f79fbdcd792aff04b2bf111de13219cd9ad56 - thanks for keeping this updated.

**Pull request merged**